lib/trino-hive-formats/src/test/java/io/trino/hive/formats/ion/TestIonFormat.java

```diff
@@ -435,6 +435,17 @@ public void testEncode()
         assertIonEquivalence(TEST_COLUMNS, page, ionText);
     }
 
+    @Test
+    public void testEncodeMixedCaseColumn()
+            throws IOException
+    {
+        List<Column> casedColumn = List.of(
+                new Column("TheAnswer", INTEGER, 0));
+
+        Page page = toPage(casedColumn, List.of(42));
+        assertIonEquivalence(casedColumn, page, "{ TheAnswer: 42 }");
+    }
```

Comment:
Nit: Based on the change, it seems that case insensitivity is only supported in the decoding process? If so, should we consider using this test to verify that case insensitivity is not supported in encoding by having a test like:
   ```
 @Test
    public void testEncodeMixedCaseColumn()
            throws IOException
    {
        List<Column> casedColumn = List.of(
                new Column("TheAnswer", INTEGER, 0));

        Page page = toPage(casedColumn, List.of(42));
        Assertions.assertThrows(AssertionError.class, () -> assertIonEquivalence(casedColumn, page, "{ theanswer: 42 }"));
    }
```

Reply:
Well I'd say it's more that the schema (the column names) is cased. data at rest is also cased. case-insensitivity is just specifying that the data at rest may match the schema even if the casing is different. 

What's there assures that the case truly matches.